### PR TITLE
Create AMQP adapter base module as a first step towards introducing Q…

### PR DESCRIPTION
**Pull request by dejanb**
…uarkus alternative

### CODE REVIEW

File: adapters/amqp-vertx-base/pom.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<!--
+    Copyright (c) 2016, 2020 Contributors to the Eclipse Foundation
```
Comment (sophokles73) [inline adapters/amqp-vertx-base/pom.xml:2]: 2021?

**Review by sophokles73 — COMMENTED** [review]
IMHO we should also move the *AmqpContext* class to the *org.eclipse.hono.adapter.amqp* package as part of this because it is not implementation specific ...

**Comment by dejanb** [comment]
@sophokles73 fixed

**Review by sophokles73 — COMMENTED** [review]
IMHO we cannot keep the *AmqpContext* class in the *org.eclipse.hono.adapter.amqp.impl* package because we would then end up with a *split package*, i.e. classes from the same package which need to be loaded from different artifacts (jars). This results in an error when loading classes form signed jars (as we  do) ...
That is why I proposed to move the class to *org.eclipse.hono.adapter.amqp* which will then be only contained in the amqp adapter base artifact, right?

**Comment by dejanb** [comment]
@sophokles73 yeah, it makes sense. Fixed

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**